### DISCUSSION TITLE
SSG with dynamic paths

### DISCUSSION THREAD

**Discussion by caguy** [solidjs/solid-start#1548] [Q&A]
Let's say I want to render all my blog post pages at build time (SSG). I would need to dynamically fetch my CMS or DB to get the slug list of all blog post to generate during the build, and then loop into those slugs to render those pages statically.

In Nextjs, this can be done with [generateStaticParams](https://nextjs.org/docs/app/api-reference/functions/generate-static-params). 

I couldn't find a way to achieve this neither in the Solid nor in the Vinxi docs, but loooking a bit how it's done under the hood, I found that this works:

In app.config.ts:
```
import { defineConfig } from "@solidjs/start/config";

const getPostSlugs = async () => { // 👈 Fetcher function
  const posts = await fetch('https://.../posts').then((res) => res.json());
  return posts.map((post) => post.slug);
} 

export default defineConfig({
  server: {
    preset: "node-server",
    prerender: { routes: [] }, // 👈 Use an empty array (or fill it with static paths) and the dynamic paths will be pushed in a hook later
    hooks: {
      async "prerender:routes"(routes) { // 👈 This hook is called just before reading the routes array from previous line
        const slugs = await getPostSlugs(); 
        slugs.forEach((slug) => routes.add(`/post/${slug}`));
      },
    },
  },
});
```

Is this the official way to SSG with dynamic paths?

**Comment by multivoltage** [comment]
How did you that?

**Comment by samerset-rmn** [comment]
Hi, any updates? :)
